examples/src/test/java/io/numaproj/numaflow/examples/source/simple/SimpleSourceTest.java

```diff
@@ -0,0 +1,78 @@
+package io.numaproj.numaflow.examples.source.simple;
+
+import io.numaproj.numaflow.sourcer.AckRequest;
+import io.numaproj.numaflow.sourcer.Message;
+import io.numaproj.numaflow.sourcer.Offset;
+import io.numaproj.numaflow.sourcer.OutputObserver;
+import io.numaproj.numaflow.sourcer.ReadRequest;
+import lombok.Data;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SimpleSourceTest {
+
+    @Data
+    static class TestObserver implements OutputObserver {
+        List<Message> messages = new ArrayList<>();
+
+        @Override
+        public void send(Message message) {
+            messages.add(message);
+        }
+    }
+
+    @Test
+    public void test_ReadAndAck() {
+        SimpleSource simpleSource = new SimpleSource();
+        TestObserver testObserver = new TestObserver();
+
+        // Read 2 messages
+        ReadRequest readRequest = Mockito.mock(ReadRequest.class);
```

Comment:
same here as the datum. It would be nice if an instance of the ReadRequest class was easy to construct instead of needing to mock it.

Reply:
The ReadRequest is an interface. Users can create a class that implements the ReadRequest interface for testing or they can mock it.